camel-core/src/main/java/org/apache/camel/processor/RestBindingAdvice.java

```diff
@@ -337,13 +337,13 @@ private void marshal(Exchange exchange, Map<String, Object> state) {
                 // only marshal if its json content type
                 if (contentType.contains("json")) {
                     jsonMarshal.process(exchange);
-                    exchange.setProperty(Exchange.OUTPUT_TYPE, new DataType("json"));
+                    setOutputDataType(exchange, new DataType("json"));
```

Comment:
would it make sense to have specific JSonDatatype and XMLDataType as they will be used a lot of time? and maybe a singleton to avoid several instanciation?